src/cript/api/api.py

```diff
@@ -85,18 +86,44 @@ class API:
     _db_schema: dict = {}
     _http_headers: dict = {}
 
-    def __init__(self, host: Union[str, None], token: [str, None]):
+    def __init__(self, host: Union[str, None] = "", token: [str, None] = "", config_file_path: Union[Path, str] = ""):
         """
-        Initialize object with host and token.
-        It is necessary to use a `with` context manager for the API
+        Initialize CRIPT API client with host and token.
+        Additionally, you can  use a config.json file and specify the file path.
+
+        !!! note "api client context manager"
+            It is necessary to use a `with` context manager for the API
 
         Examples
         --------
+        ### Create API client with host and token
         ```Python
         with cript.API('https://criptapp.org', 'secret_token') as api:
            # node creation, api.save(), etc.
         ```
 
+        ---
+
+        ### Create API client with config.json
+        `config.json`
+        ```json
+        {
+            "host": "https://criptapp.org",
+            "token": "I am token"
+        }
+        ```
+
+        `my_script.py`
+        ```python
+        from pathlib import Path
+
+        # create a file path object of where the config file is
+        config_file_path = Path(__file__) / Path('./config.json')
```

Comment:
This seems overly complicated.
I assume this to be compatible with windows?

I would recomment `os.path.join` and `os.path` for this.

Reply:
I think this is the recommended way because that way the file path would be found regardless of the operating system, and if I am not mistaking I think this it the recommended way of getting a file path.

From when I was looking into it, I think I found that `pathlib.Path` is built on top of `os.path` and is often considered a nicer interface.

I wasn't sure if I should just accept anything and just convert it under the hood to a `Path` object or not.  The user has a bunch of options they can choose from. 

I can also put all the possibilities in the documentation as well if we want

---

## Usage
### absolute path string
```python
config_file="C:\\Users\username\desktop\crip\my_file"
```

### relative path
```python
config_file_path ="./.config.json"
```

### pathlib Path object
```python
# always gets the correct file regardless of which directory the code is ran from
config_file_path = (Path(__file__) / Path('./config.json')).resolve
```

### os.path.join
```python
# always gets the correct file regardless of which directory the code is ran from
file_path = os.path.join(os.path.dirname(__file__), "config.json")
```

### os.path
```python
# always gets the correct file regardless of which directory the code is ran from
file_path = os.path.abspath(os.path.join(os.path.dirname(__file__), 'config.json'))
```

---

I think they would all have the similar levels of difficulty

## Difficulty
### `os.path.join`
```python
import os

config_path = os.path.join(os.path.dirname(__file__), 'config.json')
```

### `os.path`
```python
import os

config_path = os.path.abspath(os.path.join(os.path.dirname(__file__), 'config.json'))
```

### `pathlib.Path`
```Python
Path(__file__) / Path('./config.json')
```

Reply:
ok, I was just confused, because I saw you using `/` which is OS specific.